src/scanpy/get/_aggregated.py

```diff
@@ -388,35 +388,50 @@ def aggregate_chunk_sum_or_count_nonzero(
     ):
         # See https://docs.dask.org/en/stable/generated/dask.array.map_blocks.html
         # for what is contained in `block_info`.
-        subset = slice(*block_info[0]["array-location"][0])
-        by_subsetted = by[subset]
-        mask_subsetted = mask[subset] if mask is not None else mask
+        if chunked_axis == 0:
+            # only subset the mask and by if we need to i.e., there is chunking along the same axis as by and mask
```

Comment:
I think this makes the comments’ scopes a bit clearer?

/edit: sheesh, GitHub is bad at displaying suggestions on a diff that spans deleted lines.

```suggestion
        # only subset the `mask` and `by` if we need to,
        # i.e. when there is chunking along the same axis as by and mask
        if chunked_axis == 0:
            # See https://docs.dask.org/en/stable/generated/dask.array.map_blocks.html
            # for what is contained in `block_info`.
```